dotCMS/src/main/java/com/dotmarketing/db/LockTask/LockTaskAPI.java

```diff
@@ -0,0 +1,179 @@
+        } catch (Exception e) {
+           throw new DotRuntimeException(e);
+        } 
+    }
+
+
+    /**
+     * This method tries to aquire a lock on a specific task for the seconds specified. If a lock cannot
+     * be aquired, it will return an empty Optional
+     * 
+     * @param taskToLock
+     * @param secondsToHoldLock
+     * @return
+     */
+    Optional<LockTask> lockTask(final String taskToLock, final int secondsToHoldLock) {
+
+        final DotConnect db = new DotConnect();
+        deleteOldLocks();
+        Date future = new Date(dbNow().getTime() + (secondsToHoldLock * 1000));
+        try (Connection conn = DbConnectionFactory.getDataSource().getConnection()){
+            conn.setAutoCommit(false);
+            db.setSQL(SQLHolder.sql.DELETE_MY_LOCK)
+                .addParam(taskToLock)
+                .addParam(myServerId)
+                .loadResult(conn);
+            db.setSQL(SQLHolder.sql.INSERT_LOCK)
+                .addParam(taskToLock)
+                .addParam(myServerId)
+                .addParam(future)
+                .loadResult(conn);
+            conn.setAutoCommit(true);
+            conn.commit();
+            return Optional.of(new LockTask(
+                            db.setSQL(SQLHolder.sql.SELECT_MY_LOCK).addParam(taskToLock).addParam(myServerId).loadObjectResults()));
+
+        } catch (Exception e) {
+            Logger.debug(this.getClass(),
+                            "unable to obtain lock:" + taskToLock + "," + myServerId + "," + secondsToHoldLock, e);
+        } 
+        return Optional.empty();
+    }
+
+    
+    /**
+     * This method tries to aquire a lock on a specific task for the seconds specified. If a lock cannot
+     * be aquired, it will return an empty Optional
+     * 
+     * @param taskToLock
+     * @param secondsToHoldLock
+     * @return
+     */
+    Optional<LockTask> waitForLock(final String taskToLock, final int secondsToHoldLock) {
+        Optional<LockTask> task = lockTask(taskToLock,secondsToHoldLock);
+        long message=0;
+        while(!task.isPresent()) {
+            if(message++ % 10 ==0) {
+                Logger.info(this.getClass(), "Waiting for lock :" + taskToLock + " " + message + " seconds ");
+            }
+            Try.run(()->Thread.sleep(1000));
+            task = lockTask(taskToLock,secondsToHoldLock);
+        }
+        return task;
+        
+    }
+    
+    
+    
+    
+    
+    /**
+     * deletes expired locks
+     */
+    @CloseDBIfOpened
+    private void deleteOldLocks() {
+
+        Try.run(() -> new DotConnect().setSQL(SQLHolder.sql.DELETE_OLD_LOCKS).loadResult())
+                        .getOrElseThrow(e -> new DotRuntimeException(e));
+
+
+    }
+
+
+    /**
+     * This will unlock the given task if it is owned by this server
+     * 
+     * @param taskToUnLock
+     * @return
+     */
+    boolean unlockTask(String taskToUnLock) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'taskToUnLock' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5639225)